templates/fact.yaml.erb

```diff
@@ -1,3 +1,7 @@
 # custom fact <%= @title %>
----
-<%= @title %>: "<%= @value %>"
+<%- require 'yaml' -%>
+<%- if RUBY_VERSION =~ /^1\.8\./ -%>
+<%= YAML::dump(@facter_data) %>
+<%- else -%>
+<%= @facter_data.to_yaml(indentation:2, canonical:false, line_width:-1) %>
```

Comment:
> Can you confirm this performs properly with non structured facts?

Yes, it's working.
Examples:
```
# puppet -V                                                                                                                                                                                                                           
3.7.5                                                                                                                                                                                                                                 
# pwd                                                                                                                                                                                                                                 
/etc/facter/facts.d                                                                                                                                                                                                                   
# cat test.yaml                                                                                                                                                                                                                       
---                                                                                                                                                                                                                                   
  key1: 
    - val11
    - val12
  key2: 
    ok: val21
    okey:
      okey1: val1
      okey2: val2
      okey3:
      - test
      - 1
      - 5
# facter -v
2.4.4
# facter key1
["val11", "val12"]
# facter key2
{"ok"=>"val21", "okey"=>{"okey1"=>"val1", "okey2"=>"val2", "okey3"=>["test", 1, 5]}}
```
```
# puppet -V
4.5.3
# pwd
/opt/puppetlabs/facter/facts.d
# cat test.yaml 
---
  key1:
    - val11
    - val12
  key2:
    ok: val21
    okey:
      okey1: val1
      okey2: val2
      okey3:
      - test
      - 1
      - 5
      - 10at
  key3:
  - okey1: val1 
    okey2: 11 
# facter -v
3.3.0 (commit fc9282a2d426f7c8a742fd5b23c4cb06852eee20)
# facter key1
[
  "val11",
  "val12"
]
# facter key2
{
  ok => "val21",
  okey => {
    okey1 => "val1",
    okey2 => "val2",
    okey3 => [
      "test",
      1,
      5,
      "10at"
    ]
  }
}
# facter key3
[
  {
    okey1 => "val1",
    okey2 => 11
  }
]
# facter key2.ok
val21
```
For the correct operation of the facts necessary valid yaml format:
[link for fcater 2.1](https://docs.puppet.com/facter/2.1/custom_facts.html#structured-data-facts), [link for fcater 2.4](https://docs.puppet.com/facter/2.4/custom_facts.html#structured-data-facts), 
[link for fcater 3.3](https://docs.puppet.com/facter/3.3/custom_facts.html#structured-data-facts).
I can add 'validate_cmd' for [yaml files](https://github.com/MemberIT/abstractit-puppet/commit/612d5cd4244ff0cceae1efefff7957b5c70ef9e3), but i thin is not needed (overkill), because content in files created yaml library.
P.S. [Commit](https://travis-ci.org/MemberIT/abstractit-puppet/builds/151475847) with correct tests.